client/driver/docker.go

```diff
@@ -731,12 +786,18 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
 		PortLowerBound: d.config.ClientMinPort,
 		PortUpperBound: d.config.ClientMaxPort,
 	}
-	ss, err := exec.LaunchSyslogServer(executorCtx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to start syslog collector: %v", err)
+
+	// Only launch syslog server if we're going to use it!
+	syslogAddr := ""
+	if len(driverConfig.Logging) == 0 || driverConfig.Logging[0].Type == "syslog" {
```

Comment:
At this point it can be. In the createContainer call a few lines down it adds the default logging option if this slice is empty.